libs/jsonforms-components/src/lib/Controls/FormStepper/util/GenerateFormFields.ts

```diff
@@ -1,35 +1,107 @@
-export const resolveLabelFromScope = (scope: string) => {
+import { LabelDescription } from '@jsonforms/core';
+
+type jsonformsLabel = string | boolean | LabelDescription | undefined;
+export const labelToString = (label: jsonformsLabel, scope: string): string => {
+  if (!label) {
+    return resolveLabelFromScope(scope);
+  }
+  if (typeof label === 'object' && label !== null && 'show' in label && label.show) {
+    return label.text ? label.text : resolveLabelFromScope(scope);
+  }
+  if (typeof label === 'string') {
+    return label;
+  }
+  return '';
+};
+
+const resolveLabelFromScope = (scope: string): string => {
   // eslint-disable-next-line no-useless-escape
   const validPatternRegex = /^#(\/properties\/[^\/]+)+$/;
   const isValid = validPatternRegex.test(scope);
-  if (!isValid) return null;
+  if (!isValid) return '';
 
-  const lastSegment = scope.split('/').pop();
+  const fieldName = scope.split('/').pop();
 
-  if (lastSegment) {
-    const lowercased = lastSegment.replace(/([A-Z])/g, ' $1').toLowerCase();
+  if (fieldName) {
+    const lowercased = fieldName.replace(/([A-Z])/g, ' $1').toLowerCase();
     return lowercased.charAt(0).toUpperCase() + lowercased.slice(1);
   }
   return '';
 };
 
-export const getFormFieldValue = (scope: string, data: object) => {
-  if (data !== undefined) {
-    const pathArray = scope.replace('#/properties/', '').replace('properties/', '').split('/');
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    let currentValue: any = data;
-    for (const key of pathArray) {
-      if (currentValue[key] === undefined) {
-        return '';
-      }
-      currentValue = currentValue[key];
+export interface InputValue {
+  type: 'primitive' | 'object';
+  value?: string[][];
+}
+
+/*
+ * Convert object to an array like [[ke1, value1], [key2, value2]]
+ * Note: this handles nested objects.
+ */
+const objToArray = (obj: object): NestedStringArray => {
+  return Object.entries(obj).map(([key, value]) => {
+    if (typeof value === 'object' && !Array.isArray(value)) {
+      return [key, objToArray(value)];
+    }
+    return [key, value];
+  });
+};
+
+type NestedStringArray = string[] | NestedStringArray[];
+
+// test for ['name', 'fred']
+const isNameValuePair = (value: NestedStringArray): boolean => {
+  return Array.isArray(value) && value.length === 2 && typeof value[0] === 'string' && typeof value[1] === 'string';
+};
+
+// test for: ['name', [['first', 'fred'], ['middle', 'jolly'], ['last', 'mcguire']]];
+const isNestedValue = (value: NestedStringArray): boolean => {
+  return Array.isArray(value) && value.length === 2 && typeof value[0] === 'string' && Array.isArray(value[1]);
+};
+
+/*
+ * Convert ['name', [['first', 'fred'], ['middle', 'jolly'], ['last', 'mcguire']]]
+ * into [['first', 'fred'], ['middle', 'jolly'], ['last', 'mcguire']]
+ */
+const flatten = (arr: NestedStringArray): string[][] => {
+  console.log('flattening');
```

Comment:
Please remove debugging code.

Reply:
Hi @rstyan, just a kind reminder that the jsonform/core includes the lodash lib, which means you can use the lodash flatten, isArray, get or other operations without enlarging the lib size.